patches/server/1044-add-PlayerBrushBlockEvent.patch

```diff
@@ -0,0 +1,29 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: chmodsayshello <chmodsayshello@hotmail.com>
+Date: Wed, 1 Nov 2023 14:01:08 +0100
+Subject: [PATCH] add PlayerBrushBlockEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/BrushableBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BrushableBlockEntity.java
+index 0425151e688966442340ea1cf892aff34ffe0791..bd21dc53ab35d9665c9687e71e4cf048f18b6807 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/BrushableBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/BrushableBlockEntity.java
+@@ -125,6 +125,18 @@ public class BrushableBlockEntity extends BlockEntity {
+ 
+     private void brushingCompleted(Player player) {
+         if (this.level != null && this.level.getServer() != null) {
++            // Paper start - PlayerBrushBlockEvent
++            io.papermc.paper.event.player.PlayerBrushBlockEvent event = new io.papermc.paper.event.player.PlayerBrushBlockEvent(
++                (org.bukkit.entity.Player) player.getBukkitEntity(),
++                org.bukkit.craftbukkit.block.CraftBlock.at(this.getLevel(), this.getBlockPos()),
++                item.asBukkitCopy()
++            );
++            event.callEvent();
++            if(event.isCancelled()) {
```

Comment:
Diff can be reduced to the following. callEvent() returns a boolean, by looking into the docs you'll see that it does what you did in 3 statements.
```suggestion
+            if(new io.papermc.paper.event.player.PlayerBrushBlockEvent((org.bukkit.entity.Player) player.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(this.getLevel(), this.getBlockPos()), item.asBukkitCopy()).callEvent()) {
```